app/engine/products/WalletsProduct.ts

```diff
@@ -0,0 +1,31 @@
+import { selectorFamily, useRecoilValue } from "recoil";
+import { Engine } from "../Engine";
+import { Address } from "ton";
+import { WalletV4State } from "../sync/startWalletV4Sync";
+import { warn } from "../../utils/log";
+
+export class WalletsProduct {
+    engine: Engine;
+    #walletSelector;
+
+    constructor(engine: Engine) {
+        this.engine = engine;
+        this.#walletSelector = selectorFamily<WalletV4State | null, string>({
+            key: 'settings/spam/deny-list',
+            get: (addr) => ({ get }) => {
```

Comment:
Do we really need selector for this?

Reply:
true, we could use just a simple engine.persistence.wallets.item atom